include/ppx/window.h

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ppx_window_h
+#define ppx_window_h
+
+#include "ppx/config.h"
+
+#include <cstdint>
+
+namespace ppx {
+
+namespace grfx {
+struct SurfaceCreateInfo;
+}
+
+class Application;
+
+struct WindowSize
+{
+    uint32_t width  = 0;
+    uint32_t height = 0;
+
+    WindowSize() = default;
+    WindowSize(uint32_t width_, uint32_t height_)
+        : width(width_), height(height_) {}
+
+    bool operator==(const WindowSize&) const = default;
+};
+
+class Window
+{
+public:
+    static Window* New(Application* app);
```

Comment:
nit: pointer args should be prefixed with p followed by camel case: `Application* pApp`.

Exceptions are historical commonly used, eg: `int main(int argc, char** argv)` or commonly used patterns by almost everyone, eg: `AInputEvent* event`.

Reply:
Those are fixed in the other PR( #136 ).

Github still don't support chaining PRs ☹️

Reply:
Boo. Got it.